common/remote/rmtssh.cpp

```diff
@@ -416,27 +444,36 @@ class CFRunSSH: public CInterface, implements IFRunSSH
                 expandCmd(cmdline,i,treefrom);
                 cmdline.append('"');
             }
-            if (dryrun) 
+            if (dryrun)
                 printf("%s\n",cmdline.str());
-            else {
+            else
+            {
                 Owned<IPipeProcess> pipe = createPipeProcess();
                 if (pipe->run((verbose&&!usepssh)?"FRUNSSH":NULL,cmdline.str(),workdir,
-                    useplink, // for some reason plink needs input handle
-                    true,true)) {
+                              useplink, // for some reason plink needs input handle
+                              true,true))
+                {
                     byte buf[4096];
-                    for (;;) {
+                    for (;;)
+                    {
                         size32_t read = pipe->read(sizeof(buf),buf);
                         if (!read)
                             break;
                         outbuf.append(read,(const char *)buf);
                     }
                     retcode = pipe->wait();
                     bool firsterr=true;
-                    for (;;) {
+                    for (;;)
+                    {
                         size32_t read = pipe->readError(sizeof(buf),buf);
                         if (!read)
                             break;
-                        if (firsterr) {
+
+                        if (!retcode)
+                            retcode = -3;
```

Comment:
One more comment, it seems a little odd to only sometimes get STDERR of " /etc/profile: line 77: ulimit: max user processes: cannot modify limit: Operation not permitted". I might suspect you would always get this msg (at least in this same host/config ?) and it seems worth looking into why it is not consistent.
Maybe there is some data left in a buffer to read ?